main.ts

```diff
@@ -1,42 +1,44 @@
-const request = require('request-promise')
+import axios, { Axios } from "axios";
+import { paths, components, external } from "./schemas/server/docs/api/ref/api";
+import createClient from "openapi-fetch";
+import { Product, SearchResult } from "./types";
 
-const defaultOptions = {
-  country: 'world'
-}
+type OffOptions = {
+  country: string;
+};
+
+/** Wrapper of OFF API */
+export class OpenFoodFacts {
+  private readonly axios: Axios;
+  private readonly client: ReturnType<typeof createClient<paths>>;
 
-/** Class OFF Node API Wrappper of OFF API */
-class OFF {
   /**
    * Create OFF object
-   * @param {Object} options - Options for the OFF Object
-   * @param {string} options.country - Country for which you want to call OFF API Client
+   * @param options - Options for the OFF Object
    */
-  constructor (options = defaultOptions) {
-    this.options = options
-    this.URL = `https://${options.country}.openfoodfacts.org`
+  constructor(
+    options: OffOptions = {
+      country: "world",
+    }
+  ) {
+    const baseUrl = `https://${options.country}.openfoodfacts.org`;
```

Comment:
Not necessary for the first version, but it could be nice to switch the base URL depending on the environment variable or a constructor parameter.

```suggestion
  constructor(
    options: OffOptions
  ) {
   const {
      country = "world",
      environment = process.env.NODE_ENV ??  "dev"
    } = options;
    
    const baseUrl = `https://${country}.openfoodfacts.${environment === 'production' ? 'org' : 'net'}`;
```